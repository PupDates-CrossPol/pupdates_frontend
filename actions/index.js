export const setUserInfo = (user) => ({
  type: 'SET_USER_INFO',
  user,
})

export const setPackInfo = pack => ({
  type: 'SET_PACK_INFO',
  pack
})

export const setPackPhotos = packPhotos => ({
  type: 'SET_PACK_PHOTOS',
  packPhotos
})

export const setOtherUsers = otherUsers => ({
  type: 'SET_OTHER_USERS',
  otherUsers
})

export const setSwipeUser = swipeUser => ({
  type: 'SET_SWIPE_USER',
  swipeUser
})

export const setSwipePack = swipePack => ({
  type: 'SET_SWIPE_PACK',
  swipePack
})

export const setSwipePackPhotos = swipePhotos => ({
  type: 'SET_SWIPE_PACK_PHOTOS',
  swipePhotos
})

export const setTempUserImage = tempUserImage => ({
	type: 'SET_TEMP_USER_IMAGE',
	tempUserImage
})

export const setImageUpload = imageUpload => ({
	type: 'SET_IMAGE_UPLOAD',
	imageUpload
})

export const setModalState = modalState => ({
	type: 'SET_MODAL_STATE',
	modalState
})

export const setMatches = matches => ({
	type: 'SET_MATCHES',
	matches
})

export const setMatchesPack = matchesPack => ({
	type: 'SET_MATCHES_PACK',
	matchesPack
})

export const setMatchesPackImages = matchesImages => ({
	type: 'SET_MATCHES_PACK_IMAGES',
	matchesImages
})