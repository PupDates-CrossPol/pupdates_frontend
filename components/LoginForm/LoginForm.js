import * as React from 'react';
import { StyleSheet, Text, View, Image, TextInput, Button, TouchableOpacity, ScrollView, SafeAreaView } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';
import * as Camera from 'expo-camera';
import * as Constants from 'expo-constants';
import * as Font from 'expo-font';
import { LinearGradient } from 'expo-linear-gradient';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';

class LoginScreen extends React.Component {
  state = {
    image: null
  }

  async componentDidMount() {
    await Font.loadAsync({
      'major-mono-display': require('../../assets/fonts/MajorMonoDisplay-Regular.ttf'),
    });
  }

  render() {
    let { image } = this.state
    return (
      <SafeAreaView>
      <ScrollView style={styles.contentContainer}>
        <Image source={require('../../assets/PupDatesLogo.png')} style={styles.image}/>
        <Text style={styles.title}>PupDates</Text>
        <TextInput placeholder="Email" style={styles.input} />
        <TextInput placeholder="Password" style={styles.input}/>
        <TouchableOpacity style={styles.button} onPress={() => this.props.navigation.navigate('Home')}>
        <LinearGradient
          colors={['orange', '#c32525']}
          style={styles.linearGradient}
          onPress={() => console.log('does this work?')}
        >
            <Text style={styles.buttonText}>Login</Text>
        </LinearGradient>
          </TouchableOpacity>
        <Text>PupDates</Text>
        <Button 
          onPress={() => this.selectImg()}
          title="Add Photo"
        />
        <Button 
          onPress={() => this.takeImg()}
          title="Camera"
        />
        {image &&
            <Image source={{ uri: image }} style={{ width: 200, height: 200 }} />}
      </ScrollView>
      </SafeAreaView>
    );
  }

selectImg() {
  ImagePicker.requestCameraRollPermissionsAsync()
    .then(response => {
      if (response.granted === true) {
        ImagePicker.launchImageLibraryAsync()
          .then(response => this.setState({ image: response.uri }))
      }
    })
  }

takeImg() {
  Camera.requestPermissionsAsync()
    .then(response => {
      if (response.granted === true) {
        ImagePicker.launchCameraAsync()
          .then(response => this.setState({ image: response.uri }))
      }
    })
  }
}



const styles = StyleSheet.create({
  contentContainer: {
    paddingVertical: 20
  },

  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    marginTop: 90
  },
  image: {
    height: 140,
    width: 140,
  },
  title: {
    fontSize: 50,
    // fontFamily: 'major-mono-display'
  },
  input: {
    height: 30,
    width: '70%',
    borderColor: 'lightgrey',
    borderRadius: 50,
    // borderColor: 'rgba(33,33,33,0.81)',
    borderWidth: 1.5,
    padding: 10,
    height: 60,
    margin: 20,
    alignItems: 'center'
  },
  linearGradient: {
    width: '70%',
    borderRadius: 25,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    // borderRadius: 50,
    borderTopRightRadius: 20,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    borderTopLeftRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    color: 'black'
  },
  buttonText: {
    color: '#fff',
    fontSize: 25,
  }
});

const AppNavigator = createSwitchNavigator({
  Login: {
    screen: LoginScreen,
  },
});

export default createAppContainer(AppNavigator)