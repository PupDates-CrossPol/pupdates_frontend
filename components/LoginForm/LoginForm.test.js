import React from 'react'
import renderer from 'react-test-renderer'
import { LoginScreen as LoginForm, mapStateToProps, mapDispatchToProps } from './LoginForm'
import * as actions from '../../actions/index'

describe('<LoginForm />', () => {
  it('has one child', () => {
    const tree = renderer.create(<LoginForm />).toJSON()
    expect(tree.children.length).toBe(1)
  })
  it('renders correctly', () => {
    const tree = renderer.create(<LoginForm />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with the user, pack, pack photos, and other users', () => {
    const mockState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
      pack: [
        {
          id: 1,
          user_id: 1,
          name: 'fluffy',
        }
      ],
      packPhotos: [
        {
          id: 1,
          dog_id: 1,
          image_url: 'someaddress'
        }
      ],
      otherUsers: [
        {
          id: 2,
          first_name: 'bob',
          last_name: 'ross',
          email: 'happylittletrees',
          description: 'a painter'
        }
      ]
    }

    const expectedState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
      pack: [
        {
          id: 1,
          user_id: 1,
          name: 'fluffy',
        }
      ],
      packPhotos: [
        {
          id: 1,
          dog_id: 1,
          image_url: 'someaddress'
        }
      ],
      otherUsers: [
        {
          id: 2,
          first_name: 'bob',
          last_name: 'ross',
          email: 'happylittletrees',
          description: 'a painter'
        }
      ]
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expectedState)

  })
})

describe('mapDispatchToProps', () => {
  it('setUserInfo should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()

    const mockUser = {
      id: 1,
      first_name: 'sara',
      last_name: 'karsh',
      email: 'email',
      description: 'a string of words'
    }

    const actionToDispatch = actions.setUserInfo(mockUser)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setUserInfo(mockUser)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('setPackInfo should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()

    const mockPack = [
      {
        id: 1,
        user_id: 1,
        name: 'fluffy',
      }
    ]

    const actionToDispatch = actions.setPackInfo(mockPack)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setPackInfo(mockPack)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('setPackPhotos should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()

    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'someaddress'
      }
    ]

    const actionToDispatch = actions.setPackPhotos(mockPackPhotos)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setPackPhotos(mockPackPhotos)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('setOtherUsers should be dispatched with the correct action', () => {
    const mockDispatch = jest.fn()

    const mockOtherUsers = [
      {
        id: 2,
        first_name: 'bob',
        last_name: 'ross',
        email: 'happylittletrees@email.com',
        description: 'a guy who paints'
      }
    ]

    const actionToDispatch = actions.setOtherUsers(mockOtherUsers)
    const mappedDispatch = mapDispatchToProps(mockDispatch)
    mappedDispatch.setOtherUsers(mockOtherUsers)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  
})