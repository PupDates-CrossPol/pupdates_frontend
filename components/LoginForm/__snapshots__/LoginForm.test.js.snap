// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "marginTop": 90,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "paddingVertical": 20,
      }
    }
  >
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 140,
            "width": 140,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 50,
          }
        }
      >
        PupDates
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "lightgrey",
            "borderRadius": 50,
            "borderWidth": 1.5,
            "height": 60,
            "margin": 20,
            "padding": 10,
            "width": "90%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "lightgrey",
            "borderRadius": 50,
            "borderWidth": 1.5,
            "height": 60,
            "margin": 20,
            "padding": 10,
            "width": "90%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 20,
            "borderBottomRightRadius": 20,
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "color": "black",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4294944000,
                4290979109,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 25,
              }
            }
          >
            Login
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
